### PR TITLE
[supervisor] inflate all git repos

### PR DESCRIPTION
**Pull request by svenefftinge**
fixes [multi-repo] Shallow clones are not inflated #9021

## How to test

start a workspace on a multi-repo. E.g. https://sefftinge-362778b072.preview.gitpod-dev.com/#https://github.com/svenefftinge/multi-repo-main
And verify that the environment variable `GITPOD_REPO_ROOTS` contains all three locations (comma separated) and all repositories are fully cloned (check with `git log`).

```release-notes
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@svenefftinge: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/supervisor/pkg/supervisor/supervisor.go [hunk 1]
```diff
@@ -328,21 +328,23 @@ func Run(options ...RunOption) {
...
-
-			cmd := runAsGitpodUser(exec.Command("git", "fetch", "--unshallow", "--tags"))
-			cmd.Env = childProcEnvvars
-			cmd.Dir = cfg.RepoRoot
-			cmd.Stdout = os.Stdout
-			cmd.Stderr = os.Stderr
-			err := cmd.Run()
-			if err != nil {
-				log.WithError(err).Error("git fetch error")
+			for _, repoRoot := range strings.Split(cfg.RepoRoots, ",") {
```
Comment (akosyakov) [inline components/supervisor/pkg/supervisor/supervisor.go:331]: Is it backward compatible? If supervisor goes first?

Comment (svenefftinge) [reply components/supervisor/pkg/supervisor/supervisor.go:331]: No, is that a deployment mode, that new supervisor gets initialized with old ws-manager? It won't initialize any repositories if the new env variable is not set. Also I don't know how supervisor behaves if the env variable doesn't exist.

File: components/supervisor/pkg/supervisor/config.go [hunk 2]
```diff
@@ -460,6 +464,10 @@ func loadWorkspaceConfigFromEnv() (*WorkspaceConfig, error) {
 	if err != nil {
 		return nil, xerrors.Errorf("cannot load workspace config: %w", err)
 	}
+	//TODO remove me after deployment (backward compatibility)
```
Comment (svenefftinge) [inline components/supervisor/pkg/supervisor/config.go:467]: @akosyakov this addresses the backward compatibility issue you mentioned

**Review by akosyakov — APPROVED** [review]
lgtm and works as advertised

**Review by csweichel — COMMENTED** [review]
Change makes a ton of sense.

Got a few comments.
Also, I didn't see a CDWP fixture which actually has multiple roots. Is that because we don't have a CDWP fixture with a composite initializer, or is there something that's broken?

File: components/content-service/pkg/initializer/initializer.go [hunk 3]
```diff
@@ -535,3 +535,27 @@ func GetCheckoutLocationFromInitializer(init *csapi.WorkspaceInitializer) string
 	}
 	return ""
 }
+
+func GetCheckoutLocationsFromInitializer(init *csapi.WorkspaceInitializer) []string {
```
[the remark this answers is not in the document]
Comment (svenefftinge) [reply components/content-service/pkg/initializer/initializer.go:539]: The single (main) repo_root is unfortunately used in a few other places, so I opted at adding a second variant. So we can slowly migrate based on need.

**Comment by svenefftinge** [comment]
> Also, I didn't see a CDWP fixture which actually has multiple roots. Is that because we don't have a CDWP fixture with a composite initializer, or is there something that's broken?

Yes, the reason is I couldn'T find a git initializer but only snapshot initializers. Maybe I missed something?

**Review by csweichel — APPROVED** [review]

**Pull request merged**